Tasks/JavaToolInstallerV0/javatoolinstaller.ts

```diff
@@ -55,19 +47,78 @@ async function getJava(versionSpec: string) {
         const extractSource = buildFilePath(extractLocation, compressedFileExtension, fileNameAndPath);
         const javaFilesExtractor = new JavaFilesExtractor();
         jdkDirectory = await javaFilesExtractor.unzipJavaDownload(extractSource, compressedFileExtension, extractLocation);
-    } else { //JDK is in a local directory. Extract to specified target directory.
+    } else if ('LocalDirectory' == taskLib.getInput('jdkSourceOption', true)) { //JDK is in a local directory. Extract to specified target directory.
         console.log(taskLib.loc('RetrievingJdkFromLocalPath'));
+        cleanDestinationDirectory(cleanDestDir, extractLocation);
+
         compressedFileExtension = getFileEnding(taskLib.getInput('jdkFile', true));
         const javaFilesExtractor = new JavaFilesExtractor();
         jdkDirectory = await javaFilesExtractor.unzipJavaDownload(taskLib.getInput('jdkFile', true), compressedFileExtension, extractLocation);
+    } else if ('AptGet' == taskLib.getInput('jdkSourceOption', true)) { //Install JDK via apt-get
+        //Check that it is Linux. Apt-get does not work on all Linux distributions, however we have no way of checking which one is being used.
+        if(os.platform() === 'linux') {
+            let escapedScript = path.join(__dirname, 'aptGetJavaInstall.sh').replace(/'/g, "''");
+            fs.chmodSync(escapedScript, "777");
+
+            let scriptRunner = taskLib.tool(taskLib.which('bash', true))
```

Comment:
IIRC you can just say `taskLib.tool('bash')` here and it will find in PATH.  can also use `const` if not rebound